src/DataStructures/VectorImpl.hpp

```diff
@@ -390,3 +390,153 @@ std::ostream& operator<<(std::ostream& os,
   sequence_print_helper(os, d.begin(), d.end());
   return os;
 }
+
+/*!
+ * \ingroup DataStructuresGroup
+ * \brief Instructs Blaze to provide the appropriate vector result type after
+ * math operations. This is accomplished by specializing Blaze's type traits
+ * that are used for handling return type deduction and specifying the `using
+ * Type =` nested type alias in the traits.
+ *
+ * \param VECTOR_TYPE The vector type, which matches the type of the operation
+ * result (e.g. `DataVector`)
+ *
+ * \param BLAZE_MATH_TRAIT The blaze trait/expression for which you want to
+ * specify the return type (e.g. `AddTrait`).
+ */
+#define BLAZE_TRAIT_SPECIALIZE_BINARY_TRAIT(VECTOR_TYPE, BLAZE_MATH_TRAIT) \
+  template <>                                                              \
+  struct BLAZE_MATH_TRAIT<VECTOR_TYPE, VECTOR_TYPE> {                      \
+    using Type = VECTOR_TYPE;                                              \
+  };                                                                       \
+  template <>                                                              \
+  struct BLAZE_MATH_TRAIT<VECTOR_TYPE, VECTOR_TYPE::value_type> {          \
+    using Type = VECTOR_TYPE;                                              \
+  };                                                                       \
+  template <>                                                              \
+  struct BLAZE_MATH_TRAIT<VECTOR_TYPE::value_type, VECTOR_TYPE> {          \
+    using Type = VECTOR_TYPE;                                              \
+  }
+
+/*!
+ * \ingroup DataStructuresGroup
+ * \brief Instructs Blaze to provide the appropriate vector result type of an
+ * operator between `VECTOR_TYPE` and `COMPATIBLE`, where the operation is
+ * represented by `BLAZE_MATH_TRAIT`
+ *
+ * \param VECTOR_TYPE The vector type, which matches the type of the operation
+ * result (e.g. `DataVector`)
+ *
+ * \param COMPATIBLE the type for which you want math operations to work with
+ * `VECTOR_TYPE` smoothly (e.g. `ComplexDataVector`)
```

Comment:
Isn't the example backwards?  I.e., VECTOR_TYPE=ComplexDataVector and COMPATIBLE=DataVector?